app/src/main/java/com/eveningoutpost/dexdrip/watch/miband/MiBandService.java

```diff
@@ -190,9 +277,10 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         try {
             if (shouldServiceRun()) {
                 final String authMac = MiBand.getPersistentAuthMac();
-                String mac = MiBand.getMac();
-                MiBand.MiBandType currDevice = MiBand.getMibandType();
-                if ((currDevice != prevDeviceType) && currDevice != UNKNOWN) {
+                String mac
+                        = MiBand.getMac();
```

Comment:
Please remove the line break here.

A few lines below please change `if (!authMac.equalsIgnoreCase(mac) || authMac.isEmpty()) {` to `if (authMac.isEmpty() || !authMac.equalsIgnoreCase(mac)) {`. That's more logical and tests in the correct order.